### ISSUE TITLE
Remove uninformative link phrases like "read more" from view templates

### ISSUE THREAD

**Issue by 3dogMcNeill** [plone/Products.CMFPlone#799]
To improve accessibility we should remove uninformative link phrases from our templates, like "read more," on  folder_summary_view and summary_view. 

![readmoretext](https://cloud.githubusercontent.com/assets/7101218/9087505/1ce282d0-3b59-11e5-8934-6361fe507c42.png)

**Comment by vangheem** [comment]
How does it look when a description is provided? I don't mind having "read more..."

many sites like have a "read more..." link in listing.

**Comment by davilima6** [comment]
I agree with @vangheem it's a desirable link in most listings (and we're talking summary views here) but I also think it's often uninformative and sometimes just wrong for some content-types (eg: image, link, 3rd-party) and site editor's intentions (eg: there's no reading material inside some folders or maybe they exist because of site architecture more than because of content).

So +1 on keeping it only when summary is available - and maybe we should target specific content-types.

**Comment by 3dogMcNeill** [comment]
It's not best practices for web writing, accessibility, or SEO to have multiple links to the same thing (in this case, "read more" and the title link to the same thing—ideally, the title should be the link because it is informative and if styled correctly, is recognized as a link).

**Comment by hvelarde** [comment]
+1 on @davilima6

**Comment by 3dogMcNeill** [comment]
I don't know about other Universities, but these types of links are supposed to be removed from Penn State sites or they don't pass accessibility checks. 

http://webaim.org/techniques/hypertext/link_text
http://accessibility.psu.edu/linktext/

**Comment by hvelarde** [comment]
then may be we just need to remove the redundant links and treat that use case as something to be implemented by the theme.

**Comment by 3dogMcNeill** [comment]
I get that it's a feature that some people like. I'm concerned that if we're touting Plone as being one of the best/most accessibile CMSs available, that we're not cognizant of less-technical features that aren't compliant. In other words, I know there has been a lot of great back-end work done to be WCAG compliant and I don't mean to diminish that—I raised this issue to **complement** all of that work. Even though a site admin can easily and quickly modify the templates, I'm questioning whether they should **have** to do that or if we should be giving them a more accessible product OTB.

**Comment by enzus** [comment]
I found this blog post which could be relevant here:
https://silktide.com/i-thought-title-text-improved-accessibility-i-was-wrong

Odd but functional solution suggested by the W3C.
Although in our case it doesn't solve the problem of having two elements with the same link.

**Comment by 3dogMcNeill** [comment]
I read that. The "read more" isn't needed when the title text is descriptive and links to the same thing.

**Comment by davilima6** [comment]
Thanks @3dogMcNeill for pointing out those issues we were unaware of. I think it's also important to check how [Drupal](https://www.drupal.org/node/49428), [Word](http://www.elegantthemes.com/blog/tips-tricks/how-to-customize-the-wordpress-read-more-link-text)[Press](http://codex.wordpress.org/Template_Tags/the_content) and other CMS deal with this.

+1 for keeping the more marketing inclined `Read more` link if we take proper care:

* SEO problems: wrapping title, description and call-to-action in [one block-level link](http://html5doctor.com/block-level-links-in-html-5/)
* a11n problems: applying either W3C's solution _(eg: Read more about {obj/Title})_ or `aria-hidden` attribute for screen readers
* webwriting/pertinency problems: applying zpt conditions on item summary and content-type

**Comment by 3dogMcNeill** [comment]
@davilima6 That seems like a good solution

**Comment by tkimnguyen** [comment]
How about an option for disabling the Read More links? A variant on that particular view template?

**Comment by davilima6** [comment]
I think that like most non-permission related DOM elements we should avoid checkboxes and leave disabling to CSS or template editing.

**Comment by tkimnguyen** [comment]
IMO requiring someone to know CSS or template editing to change that seems a bit heavy handed. With a setting we can put the disagreements to rest; site admins and content editors can decide for themselves.

**Comment by tomgross** [comment]
I'm +1 for a documented recipe using a template or CSS. The templates are complex enough already another option is an additional burden to implement, document, train, maintain, test, etc.

**Comment by thet** [comment]
My point of view:  I'm tentatively +1 for removing the read more link as suggested by @3dogMcNeill and -1 on wrapping the title, description and read more link into one block level link. The block level link idea makes it more difficult to support HTML formattings in descriptions - although Plone doesn't support it OOTB, there is an add on for it. If it helps accessibility, we should provide an read more link which is only readable by screen readers as suggested by the article linked by @enzus (but add a comment into code for the obscure CSS, what it is for). Also -1 on the checkbooks idea for the reasons already given.

**Comment by RobZoneNet** [comment]
I am with @3dogMcNeill. At the very least we should aria-hidden="true" it if you think it is useful for sighted users. For the seeing impaired there really is no context for them from the screen reader point of view and they just hear "read more" so many times.

**Comment by tkimnguyen** [comment]
@RobZoneNet this could be something to tackle for the @plone/usability-team

**Comment by jensens** [comment]
In ClassicUI this is still a problem. Volto solves it.

**Comment by jensens** [comment]
Still relevant for Plone 6. Classic UI templates still use uninformative 'read more' link phrases — accessibility concern.
